src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -30,7 +30,7 @@ import scala.reflect.ClassTag
   *  @define willNotTerminateInf
   */
 @SerialVersionUID(1L)
-class ArrayDeque[A] private(var array: Array[AnyRef], var start: Int, var end: Int)
+class ArrayDeque[A] private(var array: Array[Any], var start: Int, var end: Int)
```

Comment:
@Ichoran @DarkDimius : This is interesting - was not aware of the perf implications. Reverted back: https://github.com/scala/collection-strawman/pull/49/commits/5f137d5506a7e0c1a7fbedb1974a0afd97731371